Documentation/Global/Image.rst

```diff
@@ -9,73 +9,148 @@ Image ViewHelper `<f:image>`
+to display an image with the `<f:image>` ViewHelper.
 
-Output in frontend::
+See `Example: Render the images stored in field image <https://docs.typo3.org/permalink/t3tsref:filesprocessor-example-render-image>`_
+in the TypoScript reference.
 
-   <img alt="alt text" src="typo3conf/ext/myext/Resources/Public/typo3_logo.png" width="396" height="375" />
+You can then use the :ref:`image <t3viewhelper:viewhelper-argument-typo3-cms-fluid-viewhelpers-imageviewhelper-image>`
+argument to pass the :php-short:`\TYPO3\CMS\Core\Resource\File` or
+:php-short:`\TYPO3\CMS\Core\Resource\FileReference`
+provided by the TypoScript data processor to the ViewHelper.
 
-or in backend::
+..  code-block:: html
+
+    <f:for each="{images}" as="image">
+        <f:image image="{image}" class="something" height="250" alt="Description"/>
+    </f:for>
 
-   <img alt="alt text" src="../typo3conf/ext/viewhelpertest/Resources/Public/typo3_logo.png" width="396" height="375" />
+The :abbr:`FAL (file abstraction layer)` will take care of scaling the image and
+converting it to a web format if necessary (for example a pdf to a png).
 
-Image Object
-------------
+The `<f:image>` ViewHelper generates a HTML `<img>` tag with the correct paths etc.
 
-::
+..  _typo3-fluid-image-extbase:
 
-   <f:image image="{imageObject}" />
+Images from Extbase models in Fluid
+-----------------------------------
 
-Output::
+In Extbase models you can use fields of type
+:php:`\TYPO3\CMS\Extbase\Domain\Model\FileReference` if a field may contain
+maximal one image and :php:`\TYPO3\CMS\Extbase\Persistence\ObjectStorage` if a
+field may contain several images. They can then be displayed using the
+:ref:`image <t3viewhelper:viewhelper-argument-typo3-cms-fluid-viewhelpers-imageviewhelper-image>`
+argument:
 
-   <img alt="alt set in image record" src="fileadmin/_processed_/323223424.png" width="396" height="375" />
+..  code-block:: html
 
-Inline notation
----------------
+    <f:for each="{mymodel.images}" as="image">
+        <f:image image="{image}" class="something" height="250"/>
+    </f:for>
 
-::
+..  _typo3-fluid-image-uidreference:
 
-   {f:image(src: 'EXT:viewhelpertest/Resources/Public/typo3_logo.png', alt: 'alt text', minWidth: 30, maxWidth: 40)}
+Using the UID of the database entry directly
+============================================
 
-Output::
+Using :php-short:`\TYPO3\CMS\Extbase\Domain\Model\FileReference` objects is the
+preferred way of displaying an image in Fluid. However if only the UID of
+database entry is available you can also use the `<f:image>` ViewHelper with the
+UID.
 
-   <img alt="alt text" src="../typo3temp/assets/images/f13d79a526.png" width="40" height="38" />
+If the UID of the **sys_file** table is available you can pass this reference to
+the :ref:`src <t3viewhelper:viewhelper-argument-typo3-cms-fluid-viewhelpers-imageviewhelper-src>`
+argument of the ViewHelper directly.
```

Comment:
If you're using the UID of a `sys_file` record you get lost the possibility to use the cropping feature, which is only available in `sys_file_reference` records.

Reply:
Oh, never heard of this. Can you still do scaling?

Reply:
Sure. You scale width and height with values from VH and not DB. Cropping is also possible for sys_file records, if you set the json string with crop attribute on your own: https://docs.typo3.org/other/typo3/view-helper-reference/main/en-us/Global/Image.html#viewhelper-argument-typo3-cms-fluid-viewhelpers-imageviewhelper-crop. But that's no fun to do so..

Reply:
So how shall we phrase this?

Reply:
Done